tests/integration-tests/tests/iam/test_iam.py

```diff
@@ -21,93 +21,192 @@
+                min_count=0,
             )
         )
     )
+    cluster.start()
+    wait_for_computefleet_changed(cluster, "ENABLED" if is_awsbatch else "RUNNING")
 
-    # Check all CloudFormation stacks after update
-    _check_lambda_role(cfn_client, lambda_client, cluster.name, updated_lambda_role_name, not is_awsbatch)
+    # Check new roles are attached to the resources
+    _check_roles(
+        cfn_client,
+        ec2_client,
+        iam_client,
+        lambda_client,
+        cluster.name,
+        another_head_instance_role,
+        another_compute_instance_role,
+        another_lambda_role,
+        not is_awsbatch,
+    )
+
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    if is_awsbatch:
+        timeout = (
+            120 if region.startswith("cn-") else 60
+        )  # Longer timeout in china regions due to less reliable networking
+        test_job_submission_awsbatch(
+            remote_command_executor, f"awsbsub --vcpus 2 --memory 256 --timeout {timeout} sleep 1"
+        )
+    else:
+        scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+        job_id = scheduler_commands.submit_command_and_assert_job_accepted(
+            submit_command_args={"command": "sleep 1", "nodes": 1}
+        )
+        scheduler_commands.wait_job_completed(job_id)
 
 
-def _check_lambda_role(cfn_client, lambda_client, stack_name, lambda_role_name, check_no_role_is_created):
-    """Test lambda role is attached to all Lambda functions in the stack and its substack."""
+def _check_roles(
+    cfn_client,
+    ec2_client,
+    iam_client,
+    lambda_client,
+    stack_name,
+    head_instance_role,
+    compute_instance_role,
+    lambda_role,
+    check_no_role_is_created,
+):
+    """Test roles are attached to EC2 instances and Lambda functions."""
     resources = cfn_client.describe_stack_resources(StackName=stack_name)["StackResources"]
     for resource in resources:
         resource_type = resource["ResourceType"]
         if check_no_role_is_created:
-            # If check_no_role_is_created, check that there is no role created in the stack and its substack.
+            # If check_no_role_is_created, check that there is no role created in the stack.
             assert_that(resource_type).is_not_equal_to("AWS::IAM::Role")
-        if resource_type == "AWS::CloudFormation::Stack":
-            # Recursively check substacks
-            _check_lambda_role(
-                cfn_client, lambda_client, resource["PhysicalResourceId"], lambda_role_name, check_no_role_is_created
-            )
         if resource_type == "AWS::Lambda::Function":
             # Check the role is attached to the Lambda function
             lambda_function = lambda_client.get_function(FunctionName=resource["PhysicalResourceId"])["Configuration"]
-            assert_that(lambda_role_name in lambda_function["Role"]).is_true()
+            assert_that(lambda_role in lambda_function["Role"]).is_true()
```

Comment:
nit: to check the presence of a key in a dict you could use `assert_that(dict).contains(key)`